response_operations_ui/views/accounts.py

```diff
@@ -11,33 +11,82 @@
 from response_operations_ui.controllers import uaa_controller
 from response_operations_ui.controllers.notify_controller import NotifyController
 from response_operations_ui.exceptions.exceptions import NotifyError
-from response_operations_ui.forms import CreateAccountForm, RequestAccountForm
+from response_operations_ui.forms import (
+    ChangeAccountName,
+    CreateAccountForm,
+    MyAccountOptionsForm,
+    RequestAccountForm,
+)
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 account_bp = Blueprint("account_bp", __name__, static_folder="static", template_folder="templates")
 
+form_redirect_mapper = {
+    "change_name": "account_bp.change_account_name",
+    "change_username": "account_bp.change_username",
+}
 
-@account_bp.route("/my-account", methods=["GET"])
+
+@account_bp.route("/my-account", methods=["GET", "POST"])
 @login_required
 def get_my_account():
-    try:
-        # Remove once we redisplay the 'my account' link
-        abort(404)
-        user_id = session["user_id"]
-        user_from_uaa = uaa_controller.get_user_by_id(user_id)
-        first_name = user_from_uaa["name"]["givenName"]
-        last_name = user_from_uaa["name"]["familyName"]
-        formatted_date = dates.format_datetime_to_string(user_from_uaa["passwordLastModified"], date_format="%d %b %Y")
-        user = {
-            "username": user_from_uaa["userName"],
-            "name": f"{first_name} {last_name}",
-            "email": user_from_uaa["emails"][0]["value"],
-            "password_last_changed": formatted_date,
-        }
-        return render_template("account/my-account.html", user=user)
-    except Exception as e:
-        return jsonify(e)
+    form = MyAccountOptionsForm()
+    form_valid = form.validate()
+    user_id = session["user_id"]
+    user_from_uaa = uaa_controller.get_user_by_id(user_id)
```

Comment:
This still isn't quite right.  You want the order to be:
- get form
- if POST and valid form
  - do stuff...
- do GET stuff